packages/share_plus/lib/share_plus_tizen.dart

```diff
@@ -2,23 +2,61 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:ui';
+
 import 'package:share_plus/share_plus.dart';
 import 'package:share_plus_platform_interface/share_plus_platform_interface.dart';
+import 'package:tizen_app_control/app_control.dart';
 
 /// The Tizen implementation of [SharePlatform].
-///
-/// This class disables platform overriding so that it always uses the default
-/// implementation of [SharePlatform].
-///
-/// Although [Share.disableSharePlatformOverride] is meant to be
-/// used for testing, it is required for Tizen to avoid platform interface
-/// being set to Linux's.
-/// https://github.com/fluttercommunity/plus_plugins/blob/fc6864f21cf1d1f6db47a3c938f24370362d00cd/packages/share_plus/lib/share_plus.dart#L30-L42
 class SharePlugin extends SharePlatform {
-  /// Tells [Share] to use [MethodChannelShare] as the default
-  /// instance of [SharePlatform].
+  /// Registers this class as the default instance of [SharePlatform].
   static void register() {
+    // Remove this line once https://github.com/fluttercommunity/plus_plugins/pull/574
+    // is released.
     // ignore: invalid_use_of_visible_for_testing_member
     Share.disableSharePlatformOverride = true;
+
+    SharePlatform.instance = SharePlugin();
+  }
+
+  @override
+  Future<void> share(
+    String text, {
+    String? subject,
+    Rect? sharePositionOrigin,
+  }) {
+    final bool hasSubject = subject != null && subject.isNotEmpty;
+    return AppControl(
+      operation: 'http://tizen.org/appcontrol/operation/share_text',
+      uri: hasSubject ? 'mailto:' : 'sms:',
+      launchMode: LaunchMode.group,
+      extraData: {
+        'http://tizen.org/appcontrol/data/text': text,
+        if (hasSubject) 'http://tizen.org/appcontrol/data/subject': subject,
+      },
+    ).sendLaunchRequest();
+  }
+
+  @override
+  Future<void> shareFiles(
+    List<String> paths, {
+    List<String>? mimeTypes,
```

Comment:
Is it okay if we don't set ``mimeTypes`` ?

Reply:
I referred to https://docs.tizen.org/application/native/guides/app-management/common-appcontrols and there was no such requirement.

Reply:
Previous code has following part.
```c++
    // Set mime type & image paths
    if (path_array.size() > 0) {
      ret = app_control_set_mime(handle, mime_type.c_str());
      RET_IF_ERROR(ret);

      ret = app_control_add_extra_data_array(
          handle, APP_CONTROL_DATA_PATH, path_array.data(), path_array.size());
      RET_IF_ERROR(ret);
    }
```
Just in case, I'm wondering if there is any need for the part corresponding to call ``app_control_set_mime``.